### PR TITLE
fix overwrite of HTTP headers

### PR DESCRIPTION
**Pull request by lasse-aagren**
such that downstream consumer added headers (e.g. basic auth) don't dissapear

### CODE REVIEW

File: lib/ldp/client/methods.rb [hunk 1]
```diff
@@ -69,7 +69,8 @@ def post url, body = nil, headers = {}
     logger.debug "LDP: POST [#{url}]"
     resp = http.post do |req|
       req.url munge_to_relative_url(url)
-      req.headers = default_headers.merge headers
+      extraheaders = default_headers.merge headers
+      req.headers = req.headers.merge extraheaders
```
Comment (cbeer) [inline lib/ldp/client/methods.rb]: `req.headers.merge!(default_headers).merge(headers)`?

Comment (lasse-aagren) [inline lib/ldp/client/methods.rb]: done

File: lib/ldp/client/methods.rb [hunk 2]
```diff
@@ -69,7 +69,7 @@ def post url, body = nil, headers = {}
     logger.debug "LDP: POST [#{url}]"
     resp = http.post do |req|
       req.url munge_to_relative_url(url)
-      req.headers = default_headers.merge headers
+      req.headers = req.headers.merge!(default_headers).merge(headers)
```
Comment (jcoyne) [inline lib/ldp/client/methods.rb]: I think @cbeer was suggesting something like:
```ruby 
req.headers.merge!(default_headers).merge!(headers)
```
Then you can do away with the assignment.

Comment (lasse-aagren) [inline lib/ldp/client/methods.rb]: fixed

**Comment by jcoyne** [comment]
@lasse-aagren I'm not sure that any of this work is necessary.  Take a look here:

https://github.com/cbeer/ldp/compare/basic_auth

Faraday adds basic auth headers at the very last moment. I don't think it's possible to overwrite them.

**Comment by lasse-aagren** [comment]
@jcoyne well, fact is that it works with, and doesn't without this work :)

**Comment by jcoyne** [comment]
@lasse-aagren can you write a test to demonstrate that?  It's important we have a test so that someone doesn't inadvertently revert your changes in the future.

**Comment by lasse-aagren** [comment]
@jcoyne I've pushed a test

**Comment by jcoyne** [comment]
@lasse-aagren this is looking good. Can you squash your commits?  @cbeer can you give feedback?

**Comment by lasse-aagren** [comment]
squashed

**Comment by lasse-aagren** [comment]
anything else I should do?

**Comment by jcoyne** [comment]
I'm just waiting for @cbeer to weigh in.

**Comment by lasse-aagren** [comment]
any chance this will happen?

**Comment by cbeer** [comment]
:+1:

**Pull request merged**